label_studio/io_storages/base_models.py

```diff
@@ -74,12 +74,30 @@ def _scan_and_create_links(self, link_class):
                 continue
             logger.debug(f'{self}: found new key {key}')
             data = self.get_data(key)
+
+            # predictions
+            predictions = data.get('predictions', [])
+            if predictions:
+                if 'data' not in data:
+                    raise ValueError('If you use "predictions" field in the task, '
+                                     'you must put "data" field in the task too')
+                data = data['data']
+
             with transaction.atomic():
                 task = Task.objects.create(data=data, project=self.project)
                 link_class.create(task, key, self)
                 logger.debug(f'Create {self.__class__.__name__} link with key={key} for task={task}')
                 tasks_created += 1
 
+                if not task.predictions.exists():
+                    for p in predictions:
+                        prediction = Prediction(result=p['result'], score=p.get('score'),
```

Comment:
No, we need task id inside of the each prediction

Reply:
it's better to rewrite it later with TaskSerializerBulk

Reply:
I think validating preannotations is a crucial thing, so I would move all `Prediction` instance creation in a single place to avoid code duplication in the near future